BoxContentSDK/BoxContentSDK/Models/BOXRepresentation.m

```diff
@@ -0,0 +1,77 @@
+//
+//  BOXRepresentation.m
+//  BoxContentSDK
+//
+//  Created on 7/15/16.
+//  Copyright © 2016 Box. All rights reserved.
+//
+
+#import "BOXRepresentation.h"
+#import "NSJSONSerialization+BOXAdditions.h"
+#import "BOXContentSDKConstants.h"
+
+@implementation BOXRepresentation
+
+- (instancetype)initWithJSON:(NSDictionary *)JSONResponse
+{
+    if (self = [super init]) {
+        
+        self.JSONData = JSONResponse;
+        
+        self.type = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyRepresentation
+                                                   inDictionary:JSONResponse
+                                                hasExpectedType:[NSString class]
+                                                    nullAllowed:NO];
+        
+        NSDictionary *propertiesJSON = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyProperties
+                                                                  inDictionary:JSONResponse
+                                                               hasExpectedType:[NSDictionary class]
+                                                                   nullAllowed:NO];
+        
+        self.dimensions = [propertiesJSON objectForKey:BOXAPIObjectKeyDimensions];
+        
+        self.status = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyStatus
+                                                     inDictionary:JSONResponse
+                                                  hasExpectedType:[NSString class]
+                                                      nullAllowed:NO];
+        
+        NSDictionary *detailsJSON = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyDetails
+                                                                   inDictionary:JSONResponse
+                                                                hasExpectedType:[NSDictionary class]
+                                                                    nullAllowed:NO];
+        if ([detailsJSON count] > 0) {
+            self.details = detailsJSON;
+        }
+        
+        NSDictionary *linksJSON = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyLinks
+                                                                 inDictionary:JSONResponse
+                                                              hasExpectedType:[NSDictionary class]
+                                                                  nullAllowed:NO];
+        
+        NSDictionary *contentJSON = linksJSON[BOXAPIObjectKeyContent];
+        
+        NSString *contentURLString = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyURL
+                                                             inDictionary:contentJSON
+                                                          hasExpectedType:[NSString class]
+                                                              nullAllowed:NO];
+        if (contentURLString.length > 0) {
```

Comment:
do we need this check? or will it automatically return nil if contentURLString.length == 0 or contentURLString == nil?

if we do self.contentURL = [NSURL URLWithString:contentURLString] always, it serves as initializing the property to nil accordingly as well.